Sources_v3/Models/ChannelId.swift

```diff
@@ -9,60 +9,46 @@ import Foundation
 /// It reflects channel's type and a unique id.
 ///
 public struct ChannelId: Hashable, CustomStringConvertible {
-    private static let any = "*"
     private static let separator: Character = ":"
     
     let rawValue: String
     
-    /// Returns `true` if the channel id matches "all" channels.
-    public var isAny: Bool { rawValue == ChannelType.unknown.rawValue + String(Self.separator) + Self.any }
-    
     /// Creates a new `ChannelId` value.
     ///
     /// - Parameters:
     ///     - type: A type of the channel the `ChannelId` represents.
     ///     - id: An id of the channel the `ChannelId` represents.
     ///
     public init(type: ChannelType, id: String) {
-        rawValue = type.rawValue + "\(Self.separator)" + (id.isEmpty ? ChannelId.any : id)
+        rawValue = type.rawValue + "\(Self.separator)" + id
     }
     
     init(cid: String) throws {
-        if cid == ChannelId.any {
-            self.init(type: .unknown, id: Self.any)
-            return
-        }
-        
-        if cid.contains(ChannelId.separator) {
-            let channelPair = cid.split(separator: ChannelId.separator)
-            let type = ChannelType(rawValue: String(channelPair[0]))
-            let id = String(channelPair[1])
-            self.init(type: type, id: id)
-        } else {
+        let channelPair = cid.split(separator: ChannelId.separator)
+    
+        guard
+            channelPair.count == 2,
+            !channelPair[0].replacingOccurrences(of: " ", with: "").isEmpty,
+            !channelPair[1].replacingOccurrences(of: " ", with: "").isEmpty
```

Comment:
Do I understand this correctly to be a fancy-schmancy way to check if the string isn't `"<space><space><space>"`? :)

Reply:
Yes :)
Will add a test for that